databricks/koalas/frame.py

```diff
@@ -2153,6 +2153,83 @@ def round(self, decimals=0):
             sdf = sdf.withColumn(decimal[0], F.round(decimal[0], decimal[1]))
         return DataFrame(self._internal.copy(sdf=sdf))
 
+    def duplicated(self, subset=None, keep='first'):
+        """
+        Return boolean Series denoting duplicate rows, optionally only considering certain columns.
+        Parameters
+        ----------
+        subset : column label or sequence of labels, optional
+            Only consider certain columns for identifying duplicates,
+            by default use all of the columns
+        keep : {'first', 'last', False}, default 'first'
+           - ``first`` : Mark duplicates as ``True`` except for the first occurrence.
+           - ``last`` : Mark duplicates as ``True`` except for the last occurrence.
+           - False : Mark all duplicates as ``True``.
+
+        Returns
+        -------
+        duplicated : Series
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'a': [1, 1, 1, 3],'b': [1, 1, 1, 4], 'c': [1, 1, 1, 5]})
+        >>> df
+           a  b  c
+        0  1  1  1
+        1  1  1  1
+        2  1  1  1
+        3  3  4  5
+
+        >>> df.duplicated().sort_index()
+        0    False
+        1     True
+        2     True
+        3    False
+        Name: x, dtype: bool
+
+        Mark duplicates as ``True`` except for the last occurrence.
+
+        >>> df.duplicated(keep='last').sort_index()
+        0     True
+        1     True
+        2    False
+        3    False
+        Name: x, dtype: bool
+
+        Mark all duplicates as ``True``.
+
+        >>> df.duplicated(keep=False).sort_index()
+        0     True
+        1     True
+        2     True
+        3    False
+        Name: x, dtype: bool
+        """
+        if subset is None:
+            subset = self._internal.data_columns
+        else:
+            diff = set(subset).difference(set(self._internal.data_columns))
+            if len(diff) > 0:
+                raise KeyError(','.join(diff))
+        sdf = self._sdf
+        index = self._internal.index_columns[0]
```

Comment:
Can we support multi-index or no index?
If we can, let's address here, otherwise let's check the index length beforehand.

Reply:
Hello. I don't know when is there no index.  It create __index_level_0__  index anyway .Can you give me example? Thanks.